### PR TITLE
[Storage] `az storage account create`: Add edge zone support

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**

To use this feature, we need to register AzureEdgeZone for your subscription.
Wait for 5/4 release with CRP and NRP.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

File: src/azure-cli-core/azure/cli/core/commands/parameters.py [hunk 1]
```diff
@@ -248,6 +248,8 @@ def _type(value):
 
 name_type = CLIArgumentType(options_list=['--name', '-n'], help='the primary resource name')
 
+edge_zone_type = CLIArgumentType(help='The name of edge zone.', is_preview=True)
```
Comment (MirzaSikander) [inline src/azure-cli-core/azure/cli/core/commands/parameters.py:251]: Why call this `edge_zone_type`? This should be called `edge_zone_name` imho

Comment (Juliehzl) [reply src/azure-cli-core/azure/cli/core/commands/parameters.py:251]: this type is CLI argument type, instead of extended location type.

Comment (evelyn-ys) [reply src/azure-cli-core/azure/cli/core/commands/parameters.py:251]: Suggestion: add `options_list` to ensure consistency in all command modules.

Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/commands/parameters.py:251]: Must it be in core?

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 2]
```diff
@@ -270,6 +270,7 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
                    help='The minimum TLS version to be permitted on requests to storage. '
                         'The default interpretation is TLS 1.0 for this property')
         c.argument('allow_shared_key_access', allow_shared_key_access_type)
+        c.argument('edge_zone', edge_zone_type, min_api='2020-08-01-preview')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py:273]: Does it take any string or have a allowed values' list? `microsoftrrdclab1` and `microsoftlosangeles1` from tests don't seem normal.

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py:273]: There is no way to get allowed values now unless you know service design. We will add a document link here when service team is ready.

File: src/azure-cli-core/azure/cli/core/commands/parameters.py (hunk 1 above)
Comment (Juliehzl) [reply src/azure-cli-core/azure/cli/core/commands/parameters.py:251]: There are multiple operations in different modules to use the argument, so we define it here. If not in core, each command has to define by themselves.

Comment (Juliehzl) [reply src/azure-cli-core/azure/cli/core/commands/parameters.py]: good point! updated.

**Review by evelyn-ys — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/commands/parameters.py [hunk 3]
```diff
@@ -248,6 +248,8 @@ def _type(value):
 
 name_type = CLIArgumentType(options_list=['--name', '-n'], help='the primary resource name')
 
+edge_zone_type = CLIArgumentType(options_list='--edge-zone', help='The name of edge zone.', is_preview=True)
```
Comment (qwordy) [inline src/azure-cli-core/azure/cli/core/commands/parameters.py:251]: Add a min_api?

File: src/azure-cli/azure/cli/command_modules/storage/_params.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py:273]: `is_preview` is global while `min_api` is not global. I am confused.

File: src/azure-cli-core/azure/cli/core/commands/parameters.py (hunk 3 above)
Comment (qwordy) [reply src/azure-cli-core/azure/cli/core/commands/parameters.py:251]: How do users know allowed values?

File: src/azure-cli/azure/cli/command_modules/storage/_params.py (hunk 2 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py:273]: as different RP has different api version restriction, so we cannot make it global.

File: src/azure-cli-core/azure/cli/core/commands/parameters.py (hunk 3 above)
Comment (Juliehzl) [reply src/azure-cli-core/azure/cli/core/commands/parameters.py:251]: see https://github.com/Azure/azure-cli/pull/17528#discussion_r611427729

File: src/azure-cli/azure/cli/command_modules/storage/_params.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py:273]: I mean `is_preview` should not be global.

**Review by qwordy — APPROVED** [review]

**Pull request merged**